### PR TITLE
Add release-please configuration

### LINKED ISSUE #835: add release please configuration
reflection of https://github.com/microsoftgraph/msgraph-beta-sdk-java/issues/940 on this repository. We want:
- [ ] release please header file (yaml .github/release-please.yml)
- [x] release please manifest (json .release-please-manifest.json)
- [x] release please config (json release-please-config.json)
- [x] release please workflow to trigger on main branch (.github/workflows/release-please.yml)
- [ ] install the [release please GitHub app](https://github.com/apps/release-please) to the repository
- [ ] changelog keep a changelog header and unreleased information to be removed
- [x] contributing to be updated with commit formatting instructions
- [x] version bump script to be deleted (and removed from the weekly generation process)
- [ ] files where the version needs to be updated to be ammended with the special comments for release please

### PR DESCRIPTION
**Pull request by timayabi2020**
Closes #835
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/856)

### CODE REVIEW

File: pipelines/productionBuild.yml [hunk 1]
```diff
@@ -64,14 +64,6 @@ extends:
             script: |
               echo "##vso[task.setvariable variable=JAVA_HOME]$(JAVA_HOME_11_X64)"
               echo "##vso[task.setvariable variable=PATH]$(JAVA_HOME_11_X64)\bin;$(PATH)"
-        - task: PowerShell@2
```
Comment (andrueastman) [inline pipelines/productionBuild.yml:67]: I don't think this should be removed as it acts as validation in the pipeline to ensure the version has been bumped.

File: release-please-config.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "bootstrap-sha": "9903e0c129683979234f287d59f7468ca5cd7341",
+  "exclude-paths": [
+    ".git",
+    ".idea",
+    ".github",
+    ".vscode"
+  ],
+  "release-type": "dotnet",
```
Comment (andrueastman) [inline release-please-config.json:9]: Strategy does not exist at https://github.com/googleapis/release-please/blob/72b0ab360c3d6635397e8b02f4d3f9f53932e23c/docs/customizing.md#strategy-language-types-supported

```suggestion
  "release-type": "dotnet-yoshi",
```

File: .github/workflows/release-please.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+on:
```
Comment (baywet) [inline .github/workflows/release-please.yml:1]: do we still need a workflow file? I thought the bot was handling that at this point.

File: release-please-config.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  ],
+  "release-type": "dotnet-yoshi",
+  "include-component-in-tag": false,
+  "include-v-in-tag": true,
+  "packages": {
+    ".": {
+      "package-name": "src/Microsoft.Graph.Core/",
+      "changelog-path": "CHANGELOG.md",
+      "extra-files": [
+        "src/Microsoft.Graph.Core/Microsoft.Graph.Core.csproj",
```
Comment (baywet) [inline release-please-config.json:17]: we probably need to add the comments in those files to RP knows where to make the update?

File: .github/workflows/release-please.yml (hunk 3 above)
Comment (timayabi2020) [reply .github/workflows/release-please.yml:1]: Wow. I didn't know about this. I'll go ahead and delete it

**Review by andrueastman — APPROVED** [review]
LGTM.

@baywet for final review here

File: release-please-config.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        "README.md"
```
Comment (baywet) [inline release-please-config.json:18]: I believe we don't need this entry since there's no version on the readme file

Comment (timayabi2020) [reply release-please-config.json:18]: Removing it...

**Review by baywet — APPROVED** [review]
Thanks for making all the changes!

**Pull request merged**